backend/server.js

```diff
@@ -2,28 +2,105 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import crypto from 'crypto'
+import bcrypt from 'bcrypt'
+import listEndpoints from 'express-list-endpoints'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/authAPI'
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const User = mongoose.model('User', {
+  name: {
+    type: String,
+    minlength: 3,
+    unique: true,
+    required: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+})
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const accessToken = req.header('Authorization')
+    const user = await User.findOne({ accessToken })
+
+    if (!user) {
+      throw 'Login error '
+    }
+    req.user = user
+    next()
+  } catch (err) {
+    const errorMessage = 'Try again'  
```

Comment:
This errorMessage variable does not seem to be used. You must include it in the json response on line 42 if your plan was to show that in the response when something goes wrong.
Also "Try again" might be a bit vague for an error message, it could be useful to add some more details on what triggered the error 🤓